src/controllers/colormapperjsproxy.cpp

```diff
@@ -0,0 +1,37 @@
+#include <QScriptValueIterator>
+
+#include "controllers/colormapperjsproxy.h"
+
+ColorMapperJSProxy::ColorMapperJSProxy(QScriptEngine* pScriptEngine, QMap<QRgb, QVariant> availableColors)
+    : m_pScriptEngine(pScriptEngine) {
+        m_colorMapper = new ColorMapper(availableColors);
+}
+
+QScriptValue ColorMapperJSProxy::getNearestColor(uint colorCode) {
+    QPair<QRgb, QVariant> result = m_colorMapper->getNearestColor(static_cast<QRgb>(colorCode));
+    return m_pScriptEngine->toScriptValue(result.second);
```

Comment:
We could also return an object with `red`, `green`, `blue` and `alpha` properties plus the additional `data` property here, but I don't know if there's a need for it.

Reply:
On second thought, `getNearestColor` is a misleading name for this function. It doesn't return a color, it returns a value associated with the nearest color. Maybe `getNearestValue`? We could also expose an actual `getNearestColor` function that returns the color.